jenkins/public_tests_all.Jenkinsfile

```diff
@@ -77,11 +82,14 @@ pipeline {
               }
             }
             post {
-              always {
-                catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
-                  archiveArtifacts artifacts: "flow/*tar.gz", allowEmptyArchive: true;
-                  archiveArtifacts artifacts: "flow/logs/**/*, flow/reports/**/*", allowEmptyArchive: true;
-                }
+              success {
+                archiveArtifacts artifacts: "flow/*tar.gz", allowEmptyArchive: true;
+                archiveArtifacts artifacts: "flow/logs/**/*, flow/reports/**/*", allowEmptyArchive: true;
+              }
+              failure {
+                sh "[ -f 'flow/*tar.gz' ] && cp flow/*tar.gz failures/."
```

Comment:
let's keep the current folder hier and move the failures folder to `flow/results/failures`